presto-delta/src/main/java/com/facebook/presto/delta/DeltaExpressionUtils.java

```diff
@@ -0,0 +1,224 @@
+import com.facebook.presto.common.predicate.ValueSet;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import io.delta.standalone.actions.AddFile;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.delta.DeltaColumnHandle.ColumnType.PARTITION;
+import static com.facebook.presto.delta.DeltaErrorCode.DELTA_INVALID_PARTITION_VALUE;
+import static com.facebook.presto.delta.DeltaErrorCode.DELTA_UNSUPPORTED_COLUMN_TYPE;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.Double.doubleToRawLongBits;
+import static java.lang.Double.parseDouble;
+import static java.lang.Float.floatToRawIntBits;
+import static java.lang.Float.parseFloat;
+import static java.lang.Long.parseLong;
+import static java.lang.String.format;
+
+public class DeltaExpressionUtils
+{
+    private DeltaExpressionUtils()
+    {
+    }
+
+    /**
+     * Split the predicate into partition and regular column predicates
+     */
+    public static List<TupleDomain<ColumnHandle>> splitPredicate(
+            TupleDomain<ColumnHandle> predicate)
+    {
+        Map<ColumnHandle, Domain> partitionColumnPredicates = new HashMap<>();
+        Map<ColumnHandle, Domain> regularColumnPredicates = new HashMap<>();
+
+        Optional<Map<ColumnHandle, Domain>> domains = predicate.getDomains();
+        if (domains.isPresent()) {
+            domains.get().entrySet().stream()
+                    .forEach(domainPair -> {
+                        DeltaColumnHandle columnHandle = (DeltaColumnHandle) domainPair.getKey();
+                        if (columnHandle.getColumnType() == PARTITION) {
+                            partitionColumnPredicates.put(domainPair.getKey(), domainPair.getValue());
+                        }
+                        else {
+                            regularColumnPredicates.put(domainPair.getKey(), domainPair.getValue());
+                        }
+                    });
+        }
+
+        return ImmutableList.of(
+                TupleDomain.withColumnDomains(partitionColumnPredicates),
+                TupleDomain.withColumnDomains(regularColumnPredicates));
+    }
+
+    /**
+     * Utility method that takes an iterator of {@link AddFile}s and a predicate and returns an iterator of {@link AddFile}s
+     * that satisfy the predicate (predicate evaluates to a deterministic NO)
+     */
+    public static Iterator<AddFile> iterateWithPredicate(
```

Comment:
Name this method to something more appropriate. pruneFilesUsingPartitionPredicate?

Reply:
renamed to `iterateWithPartitionPruning`.